vms/platformvm/state/state.go

```diff
@@ -1456,16 +1460,16 @@ func (s *state) loadCurrentValidators() error {
 			// Note: we don't provide [LastUpdated] here because we expect it to
 			// always be present on disk.
 		}
-		if err := parseValidatorMetadata(metadataBytes, metadata); err != nil {
+		err = parseValidatorMetadata(metadataBytes, metadata)
+		if err != nil {
 			return err
 		}
 
-		stakerTx, ok := tx.Unsigned.(txs.Staker)
-		if !ok {
-			return fmt.Errorf("expected tx type txs.Staker but got %T", tx.Unsigned)
-		}
-
-		staker, err := NewCurrentStaker(txID, stakerTx, metadata.PotentialReward)
+		staker, err := NewCurrentStaker(
+			txID,
+			stakerTx.(txs.PostDurangoStaker),
```

Comment:
this is safe because all of our StakerTxs are also PostDurangoStakerTxs and viceversa. We type assert it for each transaction.
Still ain't pretty